src/libraries/System.Runtime.Extensions/tests/System/Runtime/ProfileOptimization.cs

```diff
@@ -16,23 +16,37 @@ public class ProfileOptimizationTest : FileCleanupTestBase
         [ActiveIssue("https://github.com/dotnet/runtime/issues/31853", TestRuntimes.Mono)]
         public void ProfileOptimization_CheckFileExists(bool stopProfile)
         {
-            string profileFile = GetTestFileName();
+            // Don't put this file into the normal per-test directory, because
+            // when RemoteExecutor.Invoke returns, it disposes this class and deletes
+            // that directory. Instead just put it in temp and delete it manually after.
+            string profileFile = Path.Combine(Path.GetTempPath(), GetTestFileName());
```

Comment:
Should we use the file cleanup base instead?

Reply:
No this change is specifically to _stop_ using file cleanup base. Because RemoteExecutor.Invoke was written to dispose the test object when it returns, and if the test derives from FCTB that deletes the test's folder.

Reply:
I don't understand what's being fixed then.  Why is it ok to delete the profile file but not to delete the directory?  Your comment above doesn't say why it's not ok to delete the directory.

Reply:
The directory was getting deleted (after my change) when the Invoke returned. This test checks for its existence after Invoke returns, so it began to fail.

Reply:
In this PR, I still delete the file, but only after the test has confirmed it is present.

Reply:
> The directory was getting deleted (after my change) when the Invoke returned. 

But why?

Reply:
Because the remote process instantiates its own instance of this test class, and then disposes it. The folder name is generated from the test class name.

Reply:
> The folder name is generated from the test class name.

But it includes a randomly-generated name as well:
https://github.com/dotnet/runtime/blob/8f42760a9ad7034d194be4d1908a7a71856168e4/src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs#L40
It should not be the same between the remote process and the parent process.

Reply:
Ah, yes; clearly my previous explanation was incorrect. (Just testing my code reviewer 😛 ) TestDirectory is irrelevant.

Current directory for the test is something like `C:\git\runtime\artifacts\bin\System.Runtime.Extensions.Tests\Debug\net7.0-windows`
GetTestFileName() produces something like `ProfileOptimization_CheckFileExists_19_vviak91z`
This goes to the remote process, which passes empty string to SetProfileRoot -- because `ProfileOptimization_CheckFileExists_19_vviak91z` has no directory name. That apparently tells the CLR "write to whatever current directory is". 
RemoteExecutor now switches current directory to temp just before exiting.
As the process exits, it writes the profile, which therefore goes into temp. 
Control returns to the test, whose current directory is still `C:\git\runtime\artifacts\bin\System.Runtime.Extensions.Tests\Debug\net7.0-windows`, and it looks for the file there to no avail.

With this fix, the test uses a full path, so it passes a directory (temp in fact) to SetProfileRoot, and ultimately expects to find the file there, which it does.

Reply:
For a small increase in code coverage, I could change from GetTestFileName() to GetTestFilePath(), so the profile root is not temp, and thus the test can prove that the current directory (which is temp) is ignored. Doesn't seem worth changing today.

Reply:
Actually, it eliminates the try/catch. I'll update.

Reply:
Thanks for the explanation.  Doesn't this mean that the entirety of the fix could be changing:
```C#
string profileFile = GetTestFileName();
```
to instead be:
```C#
string profileFile = GetTestFilePath();
```
and we wouldn't need to duplicate functionality from FileCleanupTestBase into tests like this?

Reply:
yes, that's what I am doing.